### PR TITLE
Position generated trees, allowing use under -Yrangepos / IDE

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

File: src/main/scala/KindProjector.scala [hunk 1]
```diff
@@ -200,19 +200,19 @@ class KindRewriter(plugin: Plugin, val global: Global)
 
       tree match {
         case Template(parents, self, body) =>
-          Template(parents.map(transform), self, body.map(transform))
+          treeCopy.Template(tree, parents.map(transform), self, body.map(transform))
```
Comment (retronym) [inline src/main/scala/KindProjector.scala]: @non What is the purpose of this case? `super.transform(tree)` would perform:

```
treeCopy.Template(tree, transformTrees(parents), transformValDef(self), transformStats(body, tree.symbol))
```

Are you intentionally avoiding descent into the self type annotation? Or is the use of `atOwner` in `transformStats` problematic somehow?

**Comment by non** [comment]
Wow! Thanks!

File: src/main/scala/KindProjector.scala (hunk 1 above)
Comment (non) [inline src/main/scala/KindProjector.scala]: I wish I had commented -- I think there was an issue that required this but I don't recall.

Comment (non) [inline src/main/scala/KindProjector.scala]: I'll try to look into it more, and maybe test out disabling it. I was looking at that case earlier today too.

Comment (non) [inline src/main/scala/KindProjector.scala]: (Also, apologies for the quick trigger finger on merging -- I was eager to try it out.)

Comment (non) [inline src/main/scala/KindProjector.scala]: Oh! I remember.

I think there is (or was) a bug where the `parents` would not be transformed normally.

**Comment by retronym** [comment]
I've removed it in https://github.com/non/kind-projector/pull/11

**Comment by retronym** [comment]
BTW, `asPos(pos)(tree)` recursively sets positions, whereas `tree.setPos(pos)` only changes the position of `tree` itself.

**Pull request merged**